### PR TITLE
Add HTTP endpoints

### PR DESCRIPTION
**Pull request by sultan-duisenbay**
- Added HTTP endpoints for dashboard level summaries
- Added tests for HTTP endpoints
- Updated the API documentation with the new endpoints

### CODE REVIEW

File: pkg/api/README.md [hunk 1]
```diff
@@ -9,9 +9,13 @@ Exact API definitions can be found on [GitHub](https://github.com/GoogleCloudPla
 - /api/v1/dashboards - List dashboards
 - /api/v1/dashboard-groups - List dashboard groups
 - /api/v1/dashboards/{dashboard}/tabs - List a dashboard's tabs
-- /api/v1/dashboard-groups/{dashboard-group} - Lists the dashboards in a dashboard group.
+- /api/v1/dashboard-groups/{dashboard-group} - Lists the dashboards in a dashboard group
```
Comment (michelle192837) [inline pkg/api/README.md:12]: nit: Lists > List, for consistency?

File: pkg/api/README.md [hunk 2]
```diff
@@ -9,9 +9,13 @@ Exact API definitions can be found on [GitHub](https://github.com/GoogleCloudPla
...
+- /api/v1/dashboards/{dashboard}/tab-summaries - List the tab summaries for the dashboard (data rendered in dashboard view)
+- /api/v1/dashboard-groups/{dashboard-group}/dashboard-summaries - Lists the dashboard summaries for the dashboard group (data rendered in dashboard group view)
 
 ## GET
 - /api/v1/dashboards/{dashboard} - Returns a dashboard's configuration. Often empty; dashboard-level configuration is rare.
 - /api/v1/dashboards/{dashboard}/tabs/{tab}/headers - Returns the headers for a tab's grid result
 - /api/v1/dashboards/{dashboard}/tabs/{tab}/rows - Returns information on a tab's rows and the data within those rows.
+- /api/v1/dashboards/{dashboard}/tab-summaries/{tab} - Returns the summary for  a particular tab in the given dashboard
```
Comment (michelle192837) [inline pkg/api/README.md:20]: nit: double-space > single-space

**Review by michelle192837 — APPROVED** [review]
/hold

Couple of nits, but otherwise looks good!

File: pkg/api/README.md (hunk 1 above)
Comment (sultan-duisenbay) [reply pkg/api/README.md:12]: Done!

File: pkg/api/README.md (hunk 2 above)
Comment (sultan-duisenbay) [reply pkg/api/README.md:20]: Done!

**Pull request merged**